portal/server/generateApp.ts

```diff
@@ -121,21 +140,22 @@ const generateApp = () => {
 
   app.get('*', (req, res) => res.render('page-not-found.njk', { user: req.session.user }));
 
-  // error handler
-  app.use((error, req, res, next) => {
-    if (error.code === 'EBADCSRFTOKEN') {
+  const errorHandler: ErrorRequestHandler = (
+    error: {code: string, statusCode: number},
+    _req,
+    res,
+    next,
+  ) => {
+    if (error?.code === 'EBADCSRFTOKEN') {
```

Comment:
Seeing as we're keeping the optional chaining, it might be worthwhile to make the type of `error` be `{ code: string; statusCode: number } | undefined` (also notice the change from a comma to a semi-colon - running prettier in this file should handle that change, but properties in a type definition should be separated by a semi-colon)